### ISSUE TITLE
Mismatch between typing info and code

### ISSUE THREAD

**Issue by dveeden** [elastic/elasticsearch-py#1436]
**Elasticsearch version** (`bin/elasticsearch --version`):
7.6.2

**`elasticsearch-py` version (`elasticsearch.__versionstr__`)**: 
7.10.0

Please make sure the major version matches the Elasticsearch server you are running.

**Description of the problem including expected versus actual behavior**:
mypy complains after upgrading from 7.9.x to 7.10.0. The data from the `.pyi` file doesn't seem to match the `.py` file.

An example:

From `elasticsearch/client/__init__.py`:
```
    def index(self, index, body, doc_type=None, id=None, params=None, headers=None):
```

From `elasticsearch/client/__init__.pyi`:
```
    def index(
        self,
        index: Any,
        *,
        body: Any,
        doc_type: Optional[Any] = ...,
        id: Optional[Any] = ...,
        if_primary_term: Optional[Any] = ...,
        if_seq_no: Optional[Any] = ...,
        op_type: Optional[Any] = ...,
        pipeline: Optional[Any] = ...,
        refresh: Optional[Any] = ...,
        require_alias: Optional[Any] = ...,
        routing: Optional[Any] = ...,
        timeout: Optional[Any] = ...,
        version: Optional[Any] = ...,
        version_type: Optional[Any] = ...,
        wait_for_active_shards: Optional[Any] = ...,
        pretty: Optional[bool] = ...,
        human: Optional[bool] = ...,
        error_trace: Optional[bool] = ...,
        format: Optional[str] = ...,
        filter_path: Optional[Union[str, Collection[str]]] = ...,
        request_timeout: Optional[Union[int, float]] = ...,
        ignore: Optional[Union[int, Collection[int]]] = ...,
        opaque_id: Optional[str] = ...,
        params: Optional[MutableMapping[str, Any]] = ...,
        headers: Optional[MutableMapping[str, str]] = ...
    ) -> Any: ...
```

The `*` in front of the `body` argument indicates `body` can't be used as positional argument anymore, but this doesn't match what is in the `.py` file. Also the `.py` file has a different number of arguments. The info in the `.pyi` file also doesn't seem to match the docs on https://elasticsearch-py.readthedocs.io/en/7.10.0/api.html#elasticsearch.Elasticsearch.index

**Steps to reproduce**:
 Inspect `.py` and `.pyi` files.

**Comment by sethmlarson** [comment]
Thanks for opening this! I've answered this question in a different issue: https://github.com/elastic/elasticsearch-py/issues/1435#issuecomment-726828029

Basically this is to help users write better more durable code and to prepare for a future release where keyword arguments are mandatory.

**Comment by dveeden** [comment]
Thanks for your response. This is also how I interpreted this difference. Fixing our code wasn't too difficult once I found out the difference.

It would be good to make this more discoverable. One of the things I did is to use `help()` in a repl on `index()`, which basically shows the function signature and the docstring. This info was slightly misleading as this doesn't use the `.pyi` data. I would suggest to put notes/warnings/etc about upcoming changes in the docstrings where possible as this is what many people see either in a repl or in an IDE. And making sure this is covered in docs and/or changelog would also be a good step in this direction. And important piece of info that was missing for me is whether this 'mismatch' was accidental or intended.

However I also do see another possible issue: The signature from the `.pyi` has more arguments than the `.py` file. Won't that cause incompatibilities between the two signatures?
